smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/HttpProtocolTestGenerator.java

```diff
@@ -517,7 +521,17 @@ private void writeHttpHeaderAssertions(HttpMessageTestCase testCase) {
         testCase.getHeaders().forEach((header, value) -> {
             header = header.toLowerCase();
             writer.write("expect(r.headers[$S]).toBeDefined();", header);
-            writer.write("expect(r.headers[$S]).toBe($S);", header, value);
+            if (IGNORE_COMMA_SPACING.contains(header) && value.contains(",")) {
+                writer.write("""
+                    expect(
+                      r.headers[$S].replaceAll(", ", ",")
+                    ).toBe(
+                      $S.replaceAll(", ", ",")
+                    );
+                    """, header, value);
+            } else {
+                writer.write("expect(r.headers[$S]).toBe($S);", header, value);
+            }
```

Comment:
I propose that we ignore comma delimiter spacing rather than changing the upstream Smithy test assertion or our and other language implementations.

Reply:
Ah this is a pain. Although wouldn't it be easier to just add a space [here](https://github.com/smithy-lang/smithy-typescript/blob/80515513f9b4ff3e6140d335a8edfb2126b8c352/packages/middleware-compression/src/compressionMiddleware.ts#L85)?

Reply:
_should_ be fine, so I pushed a commit to do so.